File: ABC/ABC039/C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  string S;
  cin >> S;

  string s = S.substr(0,12);

  if (s == "WBWBWWBWBWBW") {
    cout << "Do" << endl;
  } else if (s == "WBWWBWBWBWWB")  {
    cout << "Re" << endl;
  } else if (s == "WWBWBWBWWBWB")  {
    cout << "Mi" << endl;
  } else if (s == "WBWBWBWWBWBW")  {
    cout << "Fa" << endl;
  } else if (s == "WBWBWWBWBWWB")  {
    cout << "So" << endl;
  } else if (s == "WBWWBWBWWBWB")  {
    cout << "La" << endl;
  } else if (s == "WWBWBWWBWBWB")  {
    cout << "Si" << endl;
  }

}

File: ABC/ABC039/C/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    
    String s = sc.next();

    sc.close();

    String S = s.substring(0,12);

    String[] patterns = {"WBWBWWBWBWBW", "WBWWBWBWBWWB", "WWBWBWBWWBWB", "WBWBWBWWBWBW", "WBWBWWBWBWWB", "WBWWBWBWWBWB", "WWBWBWWBWBWB"};
    
    String[] musicalScale = {"Do", "Re", "Mi", "Fa", "So", "La", "Si"};

    for (int i = 0; i < 7; i++) {
      if (S.equals(patterns[i])){
        System.out.println(musicalScale[i]);
        break;
      }
    }

  }

}